### PR TITLE
[CLOUD_CONTAINER_DELETION] Add logs and metrics for container deletion.

### PR DESCRIPTION
**Pull request by ankagrawal**

### CODE REVIEW

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureContainerCompactor.java [hunk 1]
```diff
@@ -67,16 +68,20 @@
    * @param cloudConfig {@link CloudConfig} object.
    * @param vcrMetrics {@link VcrMetrics} object.
    * @param azureMetrics {@link AzureMetrics} object.
+   * @throws CloudStorageException if case of any error.
    */
   public AzureContainerCompactor(AzureBlobDataAccessor azureBlobDataAccessor, CosmosDataAccessor cosmosDataAccessor,
-      CloudConfig cloudConfig, AzureCloudConfig azureCloudConfig, VcrMetrics vcrMetrics, AzureMetrics azureMetrics) {
+      CloudConfig cloudConfig, AzureCloudConfig azureCloudConfig, VcrMetrics vcrMetrics, AzureMetrics azureMetrics)
+      throws CloudStorageException {
```
Comment (lightningrob) [inline ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureContainerCompactor.java:75]: When does this throw CloudStorageException?

**Review by lightningrob — APPROVED** [review]
Approved with a few minor changes.

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureContainerCompactor.java [hunk 2]
```diff
@@ -139,6 +154,13 @@ public void compactAssignedDeprecatedContainers(Collection<? extends PartitionId
     }
   }
 
+  /**
+   * @return the latest container deletion timestamp.
+   */
+  public long getLatestContainerDeletionTimestamp() {
```
Comment (SophieGuo410) [inline ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureContainerCompactor.java:160]: minor: will this be used in the future pr?

**Review by SophieGuo410 — APPROVED** [review]
LGTM.

**Pull request merged**